src/client/threads.js

```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
     return result;
 };
 
+function listToJSON(list) {
+    if (!(list instanceof List)) {
+        return list;
+    }
+    return JSON.stringify(list.itemsArray().map(listToJSON));
+}
+
+function parseList(input) {
+    if (input.charAt(0) !== '[') {
+        return input;
+    }
+    return JSON.parse(input).map(parseList);
+}
+
+function toQueryString(obj, prefix) {
+    var str = [], k, v;
+    for(var p in obj) {
+        if (!obj.hasOwnProperty(p)) {continue;} // skip things from the prototype
```

Comment:
using `Object.keys(obj)` then using a `for` loop should be faster and should be a little more legible